docs/proposals/operator-hub.md

```diff
@@ -0,0 +1,197 @@
+# Operator Hub integration
+odo currently supports creation, listing and deletion of services created from
+the Service Catalog. With newer versions of OCP, Service Catalog is being
+deprecated in favour of Operator Hub.
+
+This proposal discusses adding similar support for Operator Hub as is currently
+available for Service Catalog. Idea is to be able to create, list and delete
+the services (basically Custom Resources in Operator framework terminology) on
+an OpenShift or Kubernetes cluster. Considering the ongoing `kclient` and DevFile
+efforts, we intend to add support for Kubernetes from day 1. However, there
+will be certain prerequisites that need to be satisfied in case of Kubernetes.
+
+Table of Contents
+- [Which Operators do we support](#which-operators-do-we-support)
+- [Installing Operators](#installing-operators)
+- [List the services/operators](#list-the-servicesoperators)
+  - [Both Service Catalog and Operator Hub are enabled](#both-service-catalog-and-operator-hub-are-enabled)
+  - [Only Service Catalog is enabled](#only-service-catalog-is-enabled)
+  - [Only Operator Hub is enabled](#only-operator-hub-is-enabled)
+  - [Kubernetes](#kubernetes)
+- [Link components with operands](#link-components-with-operands)
+- [Listing the services](#listing-the-services)
+- [Describe deployed services](#describe-deployed-services)
+- [Delete the service](#delete-the-service)
+- [Testing on Kubernetes](#testing-on-Kubernetes)
+- [Questions](#questions)
+- [GitHub issues related to this task](#github-issues-related-to-this-task)
+
+## Which Operators do we support
+Operator Hub is not the only place to find/list Operators that could be
```

Comment:
The more precise way would be to say that our goal is to use Operator Lifecycle Manager (OLM)

Reply:
OLM itself is an operator that helps manage lifecycle of the Operators. It seemed like going into the implementation detail so I didn't mention much in this doc. I'll add more info about it. :+1: